src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -241,7 +241,7 @@ public void Trace(Transaction transaction, BlockHeader block, ITxTracer txTracer
                 }
             }
 
-            UInt256 senderReservedGasPayment = noValidation ? UInt256.Zero : (ulong)gasLimit * effectiveGasPrice;
+            UInt256 senderReservedGasPayment = (ulong)gasLimit * effectiveGasPrice;
```

Comment:
We subtract fee and then refund it. In Trace(noValidation = true) we subtract zero and then refund.

Reply:
I'm quite sure we cannot do it for eth_call